experimental/GModule/GModule.jl

```diff
@@ -112,7 +112,12 @@ case that `S` is a number field and `R` is the ring of integers in `S`,
 for example `R = ZZ` and `S = QQ`)
 """
 function extension_of_scalars(M::GModule, phi::Map)
-  error("not yet ...")
+  @assert domain(phi) == base_ring(M)
+
+  d = dim(M)
+  F = free_module(codomain(phi), d)
+
+  return GModule(F, group(M), [hom(F, F, map_entries(phi, matrix(x))) for x = M.ac])
```

Comment:
nitpick: I personally prefer using `in` instead of `=` for iteration, but that's in the end a matter of taste I guess
```suggestion
  return GModule(F, group(M), [hom(F, F, map_entries(phi, matrix(x))) for x in M.ac])
```